include/itkVkComplexToComplexFFTImageFilter.h

```diff
@@ -40,22 +41,22 @@ namespace itk
  * \ingroup ITKFFT
  * \ingroup VkFFTBackend
  */
-template <typename TImage>
-class VkComplexToComplexFFTImageFilter : public ComplexToComplexFFTImageFilter<TImage>
+template <typename TInputImage, typename TOutputImage = TInputImage>
+class VkComplexToComplexFFTImageFilter : public ComplexToComplexFFTImageFilter<TInputImage, TOutputImage>
 {
 public:
   ITK_DISALLOW_COPY_AND_MOVE(VkComplexToComplexFFTImageFilter);
 
-  using InputImageType = TImage;
-  using OutputImageType = TImage;
-  static_assert(std::is_same<typename TImage::PixelType, std::complex<float>>::value ||
-                  std::is_same<typename TImage::PixelType, std::complex<double>>::value,
+  using InputImageType = TInputImage;
+  using OutputImageType = TOutputImage;
+  static_assert(std::is_same<typename TInputImage::PixelType, std::complex<float>>::value ||
+                  std::is_same<typename TInputImage::PixelType, std::complex<double>>::value,
```

Comment:
You may wish to `static_assert` for `TOutputImage` here too.  In theory, such a test will also be made when we make use of `FFTImageFilterTraits`, but the potentially redundant testing here might be appropriate nonetheless.  Similarly at other places where we `static_assert` for `TInputImage`.

Reply:
Agreed, good tradeoff of little effort for extra coverage here, just in case. I've added `static_assert` for input/output pixel type in every class.